sklearn/datasets/samples_generator.py

```diff
@@ -28,8 +27,7 @@ def _generate_hypercube(samples, dimensions, rng):
     if dimensions > 30:
         return np.hstack([_generate_hypercube(samples, dimensions - 30, rng),
                           _generate_hypercube(samples, 30, rng)])
-    out = astype(sample_without_replacement(2 ** dimensions, samples,
-                                            random_state=rng),
+    out = astype(random_state.permutation(2 ** dimensions)[:samples],
```

Comment:
Sorry for the delayed response, but even with that fix I can't seem to figure out the issue with the test case hanging. Is it possible that the permutation method is actually slower in this case for some reason?